pkgs/by-name/_0/_010editor/package.nix

```diff
@@ -0,0 +1,106 @@
+{
+  lib,
+  stdenv,
+  fetchzip,
+  fetchurl,
+  autoPatchelfHook,
+  makeWrapper,
+  makeDesktopItem,
+  cups,
+  qt5,
+  undmg,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "010editor";
+  version = "15.0";
+
+  src =
+    if stdenv.isLinux then
+      fetchzip {
+        url = "https://download.sweetscape.com/010EditorLinux64Installer${finalAttrs.version}.tar.gz";
+        hash = "sha256-mn7H0VDrBsuUYiBO3wWoCPWF/VxcdwAW6K5Yq1jCWzY=";
+      }
+    else
+      fetchurl {
+        url = "https://download.sweetscape.com/010EditorMac64Installer${finalAttrs.version}.dmg";
+        hash = "sha256-4DvTtQu1jHc7XKwCWSvSRkitvEbGTwlpVmD1UvhG0NQ=";
+      };
+
+  sourceRoot = ".";
+
+  strictDeps = true;
+  dontBuild = true;
+  dontConfigure = true;
+
+  nativeBuildInputs =
+    lib.optionals stdenv.isLinux [
+      autoPatchelfHook
+      makeWrapper
+      qt5.wrapQtAppsHook
+      stdenv.cc.cc
+    ]
+    ++ lib.optionals stdenv.isDarwin [ undmg ];
+
+  buildInputs = lib.optionals stdenv.isLinux [
+    cups
+    qt5.qtbase
+    qt5.qtwayland
+  ];
+
+  installPhase =
+    ''
+      runHook preInstall
+    ''
+    + lib.optionalString stdenv.isLinux ''
+      mkdir -p $out/opt
+      cp -ar source/* $out/opt
+
+      # Patch executable runtime
+      patchelf \
+        --set-rpath ${stdenv.cc.cc.lib}/lib:${stdenv.cc.cc.lib}/lib64 \
+        $out/opt/${finalAttrs.pname}
+
+      # Unset wrapped QT plugins since they're already included in the package,
+      # else the program crashes because of the conflict
+      makeWrapper $out/opt/${finalAttrs.pname} $out/bin/${finalAttrs.pname} \
+        --unset QT_PLUGIN_PATH
+
+      # Copy the icon and generated desktop file
+      install -D $out/opt/010_icon_128x128.png -t $out/share/icons/hicolor/128x128/apps/
+      install -D $desktopItem/share/applications/* -t $out/share/applications/
+    ''
+    + lib.optionalString stdenv.isDarwin ''
+      mkdir -p $out/Applications
+      cp -r *.app $out/Applications
+    ''
+    + ''
+      runHook postInstall
+    '';
```

Comment:
I used `mkdir` and `cp` because I was unable to use `install` to recursively copy files. Sadly, the suggested code fails to compile for me for the exact same reason.